File: lib/models/reaction.dart
class Reaction {
  const Reaction({
    required this.id,
    required this.articleId,
    required this.userId,
    required this.reactionType,
  });

  final String id;
  final String articleId;
  final String userId;
  final String reactionType;

  factory Reaction.fromJson(Map<String, dynamic> json) {
    return Reaction(
      id: json['id']?.toString() ?? '',
      articleId: json['article_id']?.toString() ?? '',
      userId: json['user_id']?.toString() ?? '',
      reactionType: json['reaction_type']?.toString() ?? 'like',
    );
  }
}

File: backend/app/schemas/reaction.py
"""Reaction-related Pydantic schemas."""

from __future__ import annotations

from datetime import datetime
from uuid import UUID
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ReactionCreate(BaseModel):
    """Payload for creating a reaction."""
    reaction_type: Literal["fire", "wow", "like"]


class ReactionResponse(BaseModel):
    """Serialized reaction response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    user_id: UUID
    article_id: UUID
    reaction_type: str
    created_at: datetime


class ReactionSummary(BaseModel):
    """Reaction summary showing counts and current user's reaction."""

    counts: dict[str, int]
    current_user_reaction: str | None = None
